mmf/common/sample.py

```diff
@@ -396,7 +396,11 @@ def to_device(
     if isinstance(device, str):
         device = torch.device(device)
 
-    if not torch.cuda.is_available():
+    # default valude of device_type is cuda
+    # since other device types such as xla can be passed
+    # falling back to cpu should only happen when device_type
+    # is set to cude but cuda is not available.
+    if not torch.cuda.is_available() and device == "cuda":
         device = torch.device("cpu")
```

Comment:
do you mind adding a warning and/or log here?

Reply:
Thanks @ytsheng , I also identified other changes. I plan to push an update later this week which should be more review ready.